dotcom-rendering/src/web/components/Liveness.importable.tsx

```diff
@@ -199,28 +199,35 @@ export const Liveness = ({
 			numHiddenBlocks > 0 ? `(${numHiddenBlocks}) ${webTitle}` : webTitle;
 	}, [numHiddenBlocks, webTitle]);
 
-	if (onFirstPage && topOfBlog) {
-		const observer = new window.IntersectionObserver(
-			([entry]) => {
-				if (entry.isIntersecting) {
-					entry.target.classList.add('in-viewport');
-					revealPendingBlocks();
-					setNumHiddenBlocks(0);
-					setShowToast(false);
-					return;
-				}
-				entry.target.classList.remove('in-viewport');
-			},
-			{
-				root: null,
-				// A margin makes more sense because the element we're
-				// observing (topOfBlog) has no height
-				rootMargin: '100px',
-			},
-		);
+	useEffect(() => {
+		if (topOfBlog) {
+			const observer = new window.IntersectionObserver(
+				([entry]) => {
+					const topOfBlogShowing = entry.isIntersecting;
 
-		observer.observe(topOfBlog);
-	}
+					if (topOfBlogShowing) {
+						entry.target.classList.add('in-viewport');
+					} else {
+						entry.target.classList.remove('in-viewport');
+					}
+
+					if (topOfBlogShowing && onFirstPage) {
+						revealPendingBlocks();
+						setNumHiddenBlocks(0);
+						setShowToast(false);
+					}
```

Comment:
Previously, this always ran, now it only runs if we're on the first page. This is the change in this PR